objects/SetupChecker/script.lua

```diff
@@ -2119,24 +2121,19 @@ function removeSpirit(params)
             found = false
             for color,obj in pairs(Global.getTable("playerTables")) do
                 if params.color ~= color then
-                    local buttons = obj.getButtons()
-                    if buttons ~= nil then
-                        for _, button in ipairs(buttons) do
-                            local label = button.label
-                            local start,_ = string.find(label, " %- ")
-                            if start ~= nil then
-                                label = string.sub(label, 1, start-1)
-                            end
-                            if name == label then
-                                replaceSpirit(obj, button.index, color)
-                                found = true
-                                break
-                            end
-                        end
-                        if found then
+                    local choices = getGainSpiritChoices(obj)
+                    for _, choice in pairs(choices) do
+                        if choice.spirit == name then
+                            replaceSpirit(obj, choice.spirit, Player[color])
+                            found = true
                             break
                         end
                     end
+                    if found then
+                        break
+                    end
+                else
+                    setupGainSpiritChoices(obj, {})
```

Comment:
what is this else case trying to solve?

Reply:
This is the equivalent of the `.clearButtons` in `Global.addSpirit`. It could perhaps do with a comment.

Reply:
ahh gotcha, since we no longer handle removing the buttons in global